### ISSUE TITLE
shoudn't Task.FromResult(0) be Task.CompletedTask

### ISSUE THREAD

**Issue by KLuuKer** [aspnet/Docs#1940]
in Docs/aspnet/security/authorization/policies.rst 

`return Task.FromResult(0);`
is used to return a Task that was completed synchronously

 shouldn't this be 
`return Task.CompletedTask;`
for a better example and less allocations (for people who just copy and past code)?

i have not checked other pages for this same issue

or maybe this should be explained\documented somewhere
[https://github.com/aspnet/Mvc/blob/fb5d92bdc02f1ed2af73fc26bbee67be20a2bb99/src/Microsoft.AspNetCore.Mvc.Core/Internal/TaskCache.cs](https://github.com/aspnet/Mvc/blob/fb5d92bdc02f1ed2af73fc26bbee67be20a2bb99/src/Microsoft.AspNetCore.Mvc.Core/Internal/TaskCache.cs)

or even better a special page for library writers to explain what people can do to squeeze out maximum performance, so we don't have to read a million blogs\lines of source code

**Comment by Rick-Anderson** [comment]
@blowdart  ?

**Comment by blowdart** [comment]
@HaoK ?

**Comment by HaoK** [comment]
Sure, Task.CompletedTask is better

**Comment by blowdart** [comment]
I never knew this. Rick do you want to update the docs? I have to now update my workshop!

**Comment by Rick-Anderson** [comment]
Sure, I'll get them updated.

**Comment by Rick-Anderson** [comment]
```
#if NET451
        public static readonly Task CompletedTask = Task.FromResult(0);
#else
        public static readonly Task CompletedTask = Task.CompletedTask;
#endif
```

@HaoK  Maybe just mention that the first time and use ``Task.FromResult(0);`` after that.

**Comment by HaoK** [comment]
Yeah I would just mention it briefly, they would have to include the shared source project too, so its not quite free to use easily

**Comment by Rick-Anderson** [comment]
fixed by #1941
